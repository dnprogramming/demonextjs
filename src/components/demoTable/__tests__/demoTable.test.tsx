import { render, screen } from "@testing-library/react";
import DemoTable from "../demoTable";
import React from "react";
import DemoContent from "@/content/demoTableData";

describe("DemoTable Component", () => {
    it("renders DemoTable description", () => {
        render(<DemoTable demoValues={DemoContent} />);
        const headerElement = screen.getByText(DemoContent[0].description);
        expect(headerElement).toBeInTheDocument();
    });


    it("renders DemoTable list items", () => {
        render(<DemoTable demoValues={DemoContent} />);
        DemoContent[0].demoData.forEach((demo) => {
                const listItemElement = screen.getByText(demo.description);
                expect(listItemElement).toBeInTheDocument();
        });
    });

    it("renders DemoTable component unchanged", () => {
        const { container } = render(<DemoTable demoValues={DemoContent} />);
        expect(container).toMatchSnapshot();
    });
});