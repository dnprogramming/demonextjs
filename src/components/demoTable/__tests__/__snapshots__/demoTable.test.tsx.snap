// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DemoTable Component renders DemoTable component unchanged 1`] = `
<div>
  <div
    id="demoTablesHeader"
  >
    <h2>
      Our Demo Tables
    </h2>
    <div
      class="demoDetails"
    >
      <div
        class="demoTabs"
      >
        <button
          class="demoTab active"
        >
          First Demo Type
        </button>
        <button
          class="demoTab "
        >
          Second Demo Type
        </button>
      </div>
      <div
        class="demoContent"
      >
        <div
          class="demoGrid"
        >
          <div
            class="demoCard"
          >
            <h2>
              First Card
            </h2>
            <p>
              This is the description for the first card.
            </p>
          </div>
          <div
            class="demoCard"
          >
            <h2>
              Second Card
              <span>
                *
              </span>
            </h2>
            <p>
              Each card is built to be displayed within the demo table system and is customizable.
            </p>
          </div>
          <div
            class="demoCard"
          >
            <h2>
              Third Card
            </h2>
            <p>
              Cards can be active or inactive, and can be marked as new to highlight recent additions.
            </p>
          </div>
        </div>
        <div
          class="demoCardNotation"
        >
          <p>
            <span>
              *
            </span>
             denotes New Demo Description!
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
